activerecord/lib/active_record/query_constraints.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  module QueryConstraints
```

Comment:
Why this module with 1 method and included in one single class exist?

Reply:
After finishing implementation of all anticipated features we will most likely inline the code into `ActiveRecord::Persistence`